include/cuco/detail/static_map.inl

```diff
@@ -358,6 +427,102 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
   }
 }
 
+template <typename Key, typename Value, cuda::thread_scope Scope, typename Allocator>
+template <typename Hash, typename KeyEqual>
+__device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::erase(
+  key_type const& k, Hash hash, KeyEqual key_equal) noexcept
+{
+  auto current_slot{initial_slot(k, hash)};
+
+  value_type const insert_pair =
+    make_pair<Key, Value>(this->get_erased_key_sentinel(), this->get_empty_value_sentinel());
+
+  while (true) {
+    auto existing_key   = current_slot->first.load(cuda::std::memory_order_relaxed);
+    auto existing_value = current_slot->second.load(cuda::std::memory_order_relaxed);
```

Comment:
```suggestion
    static_assert(sizeof(Key) == sizeof(atomic_key_type));
    static_assert(sizeof(Value) == sizeof(atomic_mapped_type));
    // TODO: Replace reinterpret_cast with atomic ref when available.
    value_type slot_contents = *reinterpret_cast<value_type const*>(current_slot);
```
We can do a naive load instead of two atomic loads with `reinterpret_cast`. `static_assert` can make sure we are not breaking anything.